src/journal/journalctl.c

```diff
@@ -1473,6 +1473,14 @@ static int list_boots(sd_journal *j) {
                         return log_error_errno(r, "Failed to set JSON field name of column 0: %m");
         }
 
+        r = table_set_sort(table, (size_t) 0);
+        if (r < 0)
+                return r;
+
+        r = table_set_reverse(table, 0, arg_reverse);
+        if (r < 0)
+                return r;
```

Comment:
our coding style says: functions should either log about all their errors, or about none. the errors we see in `list_boots()` are all logged about. But here you propagate the errors from two functions as-is without logging. hence two options:

1. continue propagating these errors, but start logging about these errors, i.e. do `return log_error_errno(r, …` here instead of just `return r;`
2. just eat up the error, cast the return value to `(void)`. i.e. don't log and don't propagate. We do this in some other cases like this. i.e. stuff whose only purpose is tweak output a bit is OK to fail silently goes the thinking behind it. 

which of the two options you pick is up to you: we have prior art for both ways to handle this in our codebase. The only option that it isn't OK here is to propagate the error and not log about it.

Reply:
similar logic applies to handling of table_set_json_field_name() errors btw. there you chose to log+propagate, which is OK. (but in fact I'd lso be OK if you'd switch to no-log+no-propagate.)

Reply:
I see, thanks. I guess I got confused how it's handled in https://github.com/systemd/systemd/blob/main/src/analyze/analyze-blame.c#L34, will update it.